astropy/io/fits/hdu/table.py

```diff
@@ -910,11 +910,19 @@ def _writedata_by_row(self, fileobj):
         for idx in range(len(self.data)):
             for field in fields:
                 item = field[idx]
+                field_width = None
 
                 if field.dtype.kind == 'U':
+                    i = field.dtype.str.index(field.dtype.kind)
+                    field_width = int(field.dtype.str[i+1:])
```

Comment:
This seems like a good utility to have especially if other code parts may be affected as well. Could you wrap these two lines in an utility and either put it in `astropy.utils` (haven't checked where it would fit best) or in `astropy.io.fits.utils`?

On the other hand there is also a `dtype.itemsize` attribute which might be more robust than parsing the string?!